src/fem/finite_element.hh

```diff
@@ -299,7 +308,7 @@ public:
      * @param p    Point of evaluation.
      * @param comp Number of vector component.
      */
-    virtual double basis_value(const unsigned int i,
+    double basis_value(const unsigned int i,
```

Comment:
Rename and reimplement to shape_value. Same for basis_grad. Called just once per assembly loop so until we need to call it many times (in XFEM) we can live with slightly reduced efficiency.

Reply:
Done.